pulsar/connector/consumer.go

```diff
@@ -1,19 +1,74 @@
 package connector
 
 import (
+	"fmt"
+	"time"
+
 	"github.com/apache/pulsar-client-go/pulsar"
 )
 
 type createConsumerOptions struct {
-	Topic            string
-	Topics           []string
-	SubscriptionName string
-	MessageChannel   chan pulsar.ConsumerMessage
+	Topic                TopicName
+	Topics               []string
+	SubscriptionName     string
+	MaxDeliveryAttempts  uint32
+	RedeliveryDelay      time.Duration
+	MessageChannel       chan pulsar.ConsumerMessage
+	DefaultBackoffPolicy bool
+	BackoffPolicy        pulsar.NackBackoffPolicy
+}
+
+func (pot *createConsumerOptions) defaults() {
+	if pot.MaxDeliveryAttempts == 0 {
+		pot.MaxDeliveryAttempts = uint32(GetClientConfig().MaxDeliveryAttempts)
+	}
+	if pot.RedeliveryDelay == 0 {
+		pot.RedeliveryDelay = time.Duration(GetClientConfig().RedeliveryDelaySeconds)
+	}
+}
+
+func (opt *createConsumerOptions) validate() error {
+	if opt.Topic == "" && len(opt.Topics) == 0 {
```

Comment:
we need to validate we didn't set them both as well...

Reply:
done